src/Native/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+#include "pal_interfaceaddresses.h"
+#include "pal_maphardwaretype.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/sysctl.h>
+
+static_assert(HAVE_AF_PACKET || HAVE_AF_LINK, "System must have AF_PACKET or AF_LINK.");
```

Comment:
Nit: it's more typical to do this sort of check--i.e. ensuring that at least one of a set of macros is defined as `1` with an `#if/#elif/#else #error` chain. For example:

```
#if HAVE_AF_PACKET
...
#elif HAVE_AF_LINK
...
#else
#error System must have AF_PACKET or AF_LINK.
#endif
```